packages/api/src/command/webhook/webhook.ts

```diff
@@ -133,78 +147,112 @@ export const processRequest = async (
-            webhookRequestId: webhookRequest.id,
-            webhookUrl,
-            context: `webhook.processRequest.updateStatus.failed`,
-            error: err2,
-          },
-        });
-      }
+      await Promise.all([
+        updateWhRequestWithError(error, webhookRequest.id, webhookUrl, status),
+        updateWhStatusWithError(error, webhookRequest.id, webhookUrl, settings.id),
+      ]);
       sendAnalytics(status);
-
-      let webhookStatusDetail;
-      if (error instanceof WebhookError) {
-        webhookStatusDetail = errorToWhStatusDetails(error);
-      } else {
-        log(`Unexpected error testing webhook`, error);
-        webhookStatusDetail = `Internal error: ${error.message}`;
-      }
-      try {
-        // update the status of the webhook endpoint on the cx's settings table
-        await updateWebhookStatus({
-          cxId: settings.id,
-          webhookEnabled: false,
-          webhookStatusDetail,
-        });
-      } catch (err2) {
-        log(`Failed to store failure state on WH settings: ${errorToString(err2)}`);
-        capture.error(err2, {
-          extra: {
-            webhookRequestId: webhookRequest.id,
-            webhookUrl,
-            context: `webhook.processRequest.updateStatus.details`,
-            error: err2,
-          },
-        });
-      }
     }
   }
   return false;
-};
+}
+
+/**
+ * Updates the individual Webhook (WH) request status.
+ */
+async function updateWhRequestWithError(
+  error: unknown,
+  webhookRequestId: string,
+  webhookUrl: string,
+  status: WebhookRequestStatus
+) {
+  try {
+    const detail =
+      error instanceof WebhookError ? errorToWhStatusDetails(error) : errorToString(error);
+    const httpStatus = error instanceof WebhookError ? error.additionalInfo.httpStatus : 500;
+    await updateWebhookRequest({
+      id: webhookRequestId,
+      status,
+      statusDetail: detail,
+      requestUrl: webhookUrl,
+      httpStatus,
+    });
+  } catch (err2) {
+    log(`Failed to store failure state on WH log: ${errorToString(err2)}`);
+    capture.error(err2, {
+      extra: {
+        webhookRequestId: webhookRequestId,
+        webhookUrl,
+        context: `webhook.processRequest.updateStatus.failed`,
+        error: err2,
+      },
+    });
+  }
+}
+
+/**
+ * Updates the Customer's Webhook status on settings.
+ */
+async function updateWhStatusWithError(
+  error: unknown,
+  webhookRequestId: string,
+  webhookUrl: string,
+  settingsId: string
+) {
+  let webhookStatusDetail;
+  if (error instanceof WebhookError) {
+    webhookStatusDetail = errorToWhStatusDetails(error);
+  } else {
+    log(`Unexpected error testing webhook: ${errorToString(error)}`);
+    webhookStatusDetail = `Internal error`;
+  }
+  try {
+    // update the status of the webhook endpoint on the cx's settings table
+    await updateWebhookStatus({
+      cxId: settingsId,
+      webhookEnabled: false,
+      webhookStatusDetail,
+    });
+  } catch (err2) {
```

Comment:
nit: error2 to be consistent in the function with naming of errors

Reply:
I just copied the code, but I can edit to `error`